ivy/func_wrapper.py

```diff
@@ -528,6 +697,38 @@ def _to_float_array(x):
     return _integer_arrays_to_float
 
 
+def casting_modes_ops(fn):
+    @functools.wraps(fn)
+    def method(*args, **kwargs):
+        # we first check if it has unsupported/supported dtypes uniquely added to it
+        intersect = set(ivy.function_unsupported_dtypes(fn)).difference(
```

Comment:
Wouldn't it be better if we just set `intersect = ivy.function_unsupported_dtypes(fn)`? Because `args` can never have an array with `invalid_dtypes` but `kwargs['dtype']` could?

Reply:
It's for a different reason, we don't apply the operation on functions that don't have the unsupported dtype attributes applied. The `function_unsupported_dtypes` returns not just the specified dtypes , but also the invalid dtypes set for that framework